builder/azure/arm/builder.go

```diff
@@ -86,23 +86,45 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 	var steps []multistep.Step
 
 	if b.config.OSType == constants.Target_Linux {
-		steps = []multistep.Step{
-			NewStepCreateResourceGroup(azureClient, ui),
-			NewStepValidateTemplate(azureClient, ui, Linux),
-			NewStepDeployTemplate(azureClient, ui, Linux),
-			NewStepGetIPAddress(azureClient, ui),
-			&communicator.StepConnectSSH{
-				Config:    &b.config.Comm,
-				Host:      lin.SSHHost,
-				SSHConfig: lin.SSHConfig(b.config.UserName),
-			},
-			&packerCommon.StepProvision{},
-			NewStepGetOSDisk(azureClient, ui),
-			NewStepPowerOffCompute(azureClient, ui),
-			NewStepCaptureImage(azureClient, ui),
-			NewStepDeleteResourceGroup(azureClient, ui),
-			NewStepDeleteOSDisk(azureClient, ui),
-		}
+		if b.config.ImageUri == "" {
+			// use marketplace template
```

Comment:
I think the better way to handle this is to transform the template with any necessary additions via the StepDeployTemplate call (or perhaps another step).

It's a lot of code to copy-n-paste for a small JSON change. Likewise, there's a lot of code already duplicated to handle Windows vs. Linux.